### ISSUE TITLE
Default action timeouts are too long

### ISSUE THREAD

**Issue by puehringer** [datavisyn/github-workflows#21]
Our default timeouts (6h) are too long for most jobs. See https://github.com/datavisyn/patentbay_product/actions/runs/5733724589, where it waited for 3h+. 

This is quite costly, and we should change that to a reasonable number.

**Comment by puehringer** [comment]
@dvviktordelev It happened again as it was stuck on restart pods ([here](https://github.com/datavisyn/infrastructure-k8s/actions/runs/5745699898/job/15574086425)): 

![Image](https://github.com/datavisyn/github-workflows/assets/51900829/7b11b21d-277a-4eb0-b4ea-ace3a2877919)

**Comment by dvviktordelev** [comment]
We can set timeout-minutes for jobs or specific steps. No global or workflow-wide option available.
Should we set timeout-minutes: 60 to the deploy jobs?

**Comment by puehringer** [comment]
Yeah, preferably on the workflow level I think. Can you check if that's passable as input? I can imagine us setting it to 30min, but you can pass 60min if required for example.

**Comment by dvviktordelev** [comment]
new PR https://github.com/datavisyn/github-workflows/pull/136
